plugins/readtimer.koplugin/main.lua

```diff
@@ -0,0 +1,95 @@
+
+local Font = require("ui/font")
+local InfoMessage = require("ui/widget/infomessage")
+local InputDialog = require("ui/widget/inputdialog")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local T = require("ffi/util").template
+local _ = require("gettext")
+
+local ReadTimer = WidgetContainer:new{
+    name = "readtimer",
+    time = 0,  -- The expected time of alarm if enabled, or 0.
+}
+
+function ReadTimer:init()
+    self.alarm_callback = function()
+        if self.time == 0 then return end -- How could this happen?
+        self.time = 0
+        UIManager:show(InfoMessage:new{
+            text = T(_("Read timer alarm\nTime's up. It's %1 now."), os.date("%c")),
+            timeout = 10,
+        })
+    end
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function ReadTimer:scheduled()
+    return self.time ~= 0
+end
+
+function ReadTimer:remainingMinutes()
+    if self:scheduled() then
+        return os.difftime(self.time, os.time()) / 60
+    else
+        return math.huge
+    end
+end
+
+function ReadTimer:unschedule()
+    if self:scheduled() then
+        UIManager:unschedule(self.alarm_callback)
+        self.time = 0
+    end
+end
+
+function ReadTimer:addToMainMenu(tab_item_table)
+    table.insert(tab_item_table.plugins, {
+        text = _("Read timer"),
+        callback = function()
+            local description = _("When will the countdown timer alarm?")
```

Comment:
I feel like this description text is kind of redundant. The title itself is enough in this case.